src/main/java/org/jmhsrobotics/frc2023/subsystems/Drivetrain.java

```diff
@@ -248,6 +261,33 @@ public void resetOdometry(Pose2d pose) {
 		);
 		// spotless:on
 	}
+	public SequentialCommandGroup createRamseteCommand(Trajectory path) {
+		final DifferentialDriveKinematics driveKinematics = new DifferentialDriveKinematics(
+				AutoConstants.trackWidthMeters);
+
+		return new InstantCommand(() -> {
+			this.isInRamsete = true;
+		}).andThen(new RamseteCommand(path, this::getPose, new RamseteController(),
+				new SimpleMotorFeedforward(AutoConstants.trajectoryFeedForwardVolts,
+						AutoConstants.trajectoryFeedForwardVoltSecondsPerMeter,
+						AutoConstants.trajectoryFeedForwardVoltSecondsSquaredPerMeter),
+				driveKinematics, this::getWheelSpeeds,
+				new PIDController(AutoConstants.trajectorykP, AutoConstants.trajectorykI, AutoConstants.trajectorykD),
+				new PIDController(AutoConstants.trajectorykP, AutoConstants.trajectorykI, AutoConstants.trajectorykD),
+				this::tankDriveVolts, this)).andThen(new InstantCommand(() -> {
+					this.isInRamsete = false;
+				}));
+	}
```

Comment:
This code should not be stored inside of the drivetrain class. If you decide to use a Ramsete command the definitions and creation should be within the command not drivetrain. 

The mode switching for isRamsete to switch between voltage and duty cycle control is mentioned in a another comment.